internal/stacks/stack_manager.go

```diff
@@ -940,61 +948,73 @@ func (s *StackManager) runFirstTimeSetup(options *types.StartOptions) (messages
-				contractLocation = contractDeploymentResult.DeployedContract.Location
-			}
-			options := make(map[string]interface{})
-			if s.Stack.CustomPinSupport {
-				options["customPinSupport"] = true
-			}
-
-			newConfig.Namespaces.Predefined[0].Multiparty = &types.MultipartyConfig{
-				Enabled: true,
-				Org:     orgConfig,
-				Node: &types.NodeConfig{
-					Name: member.NodeName,
-				},
-				Contract: []*types.ContractConfig{
-					{
-						Location:   contractLocation,
-						FirstEvent: "0",
-						Options:    options,
-					},
-				},
-			}
-		}
-
-		if err := s.patchFireFlyCoreConfigs(configDir, member, newConfig); err != nil {
-			return messages, err
-		}
+    if s.Stack.MultipartyEnabled {
+        if s.Stack.ContractAddress == "" {
+            s.Log.Info("deploying FireFly smart contracts")
+            contractDeploymentResult, err = s.blockchainProvider.DeployFireFlyContract()
+            if err != nil {
+                return messages, err
+            }
+        }
+    }
+
+    for _, member := range s.Stack.Members {
+        orgConfig := s.blockchainProvider.GetOrgConfig(s.Stack, member)
+
+        // Default to "0" if FirstEvent is not provided
+        firstEvent := "0"
+        if options.FirstEvent != "" {
+            firstEvent = options.FirstEvent
+        }
+
+        var contractLocation interface{}
+        if s.Stack.ContractAddress != "" {
+            contractLocation = map[string]interface{}{
+                "address": s.Stack.ContractAddress,
+            }
+        } else {
+            contractLocation = contractDeploymentResult.DeployedContract.Location
+        }
+
+        optionsMap := make(map[string]interface{})
+        if s.Stack.CustomPinSupport {
+            optionsMap["customPinSupport"] = true
+        }
+
+        newConfig := &types.FireflyConfig{
+            Namespaces: &types.NamespacesConfig{
+                Default: "default",
+                Predefined: []*types.Namespace{
+                    {
+                        Name:        "default",
+                        Description: "Default predefined namespace",
+                        Plugins:     []string{"database0", "blockchain0", "dataexchange0", "sharedstorage0"},
+                        Multiparty: &types.MultipartyConfig{
+                            Enabled: true,
+                            Org:     orgConfig,
+                            Node: &types.NodeConfig{
+                                Name: member.NodeName,
+                            },
+                            Contract: []*types.ContractConfig{
+                                {
+                                    Location:   contractLocation,
+                                    FirstEvent: firstEvent, // Now configurable
+                                    Options:    optionsMap,
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        }
```

Comment:
We do not want to replace the whole config - only the contract multiparty config section